public/js/actions/AtomActions/getAudioPageData.js

```diff
@@ -0,0 +1,82 @@
+import {fetchPageData, getByPath} from '../../services/capi';
+import {logError} from '../../util/logger';
+import {updateAtom} from "./updateAtom";
+
+
+export function getAudioPageData (url, atom) {
+  const path = new URL(url).pathname;
+  const pathNoLeadingSlash = path.substring(1);
+
+  return dispatch => {
+    dispatch(requestAudioPageData());
+    return getByPath(pathNoLeadingSlash)
+      .then(res => {
+        dispatch(receiveAudioPageData(res));
+        const updatedAtom = addDataToAtom(res, atom);
+        dispatch(updateAtom(updatedAtom));
+        return res;
+      })
+      .catch(error => dispatch(errorReceivingAudioPageData(error)))
+  }
+}
+
+function extractFields (audioPage) {
+  let audioEl = audioPage.elements.find(el => el.type === "audio");
+  let seriesTag = audioPage.tags.find(tag => tag.type === "series" && tag.podcast);
+  let {subscriptionUrl, googlePodcastsUrl, spotifyUrl} = seriesTag.podcast;
```

Comment:
Might be worth adding some defensive checks to this and subsequent references to found properties e.g. `let { a, b, c } = seriesTag ? seriesTag.podcast : undefined` or some such.

Reply:
Then same for `audioEl.id` and `audioEl.assets[0].file`